Added documentation for `ActiveRecord::Relation#pick` [skip ci] (#48404)
rails/rails: guides/source/active_record_querying.md

[`pluck`]: https://api.rubyonrails.org/classes/ActiveRecord/Calculations.html#method-i-pluck

### `pick`

[`pick`][] can be used to query single or multiple columns from the underlying table of a model. It accepts a list of column names as an argument and returns the first one of the specified column values ​​with corresponding data type.

> pick is a short-hand for `relation.limit(1).pluck(*column_names).first`, which makes it possible to replace code like:

Author: i will handle this.

> pick is a short-hand for `relation.limit(1).pluck(*column_names).first`, which makes it possible to replace code like:

Author: I fixed https://github.com/rails/rails/pull/48404/files#diff-77236751c3ab97b753f641ae6e10445de7662826f0176c68b5176feec92d0a1bR2263

> pick is a short-hand for `relation.limit(1).pluck(*column_names).first`, which makes it possible to replace code like:

Author: I also created a PR to change the `ActiveRecord::Relation#pluck` guide.

https://github.com/rails/rails/pull/48465

```ruby
Customer.limit(1).pluck(:id).first
```

with:

```ruby
Customer.pick(:id)
```

> 

Author: i fixed https://github.com/rails/rails/pull/48404/files#diff-77236751c3ab97b753f641ae6e10445de7662826f0176c68b5176feec92d0a1bR2275

[`pluck`]: https://api.rubyonrails.org/classes/ActiveRecord/Calculations.html#method-i-pluck

### `pick`

> [`pick`][] can be used to query single or multiple columns from the underlying table of a model. It accepts a list of column names as an argument and returns the first one of the specified column values ​​with corresponding data type.

Reviewer: I think this was copied and modified from the documentation of `pluck`.
The [explanation in the API docs](https://edgeapi.rubyonrails.org/classes/ActiveRecord/Calculations.html#method-i-pick) is a lot clearer to me.
What do you think about using some of the API docs wording?
For example:
```suggestion
[`pick`][] can be used to pick the value(s) from the named column(s) in the current relation. This is primarily useful when you have a relation that’s already narrowed down to a single row. It accepts a list of column names as an argument and returns the first one of the specified column values ​​with corresponding data type. 
```

PS. I'd also accept a seperate PR to reword the first paragraph of `pluck` to something more similar to the API docs.

Author: sounds good!
Indeed, I would change this as I thought the wording in the API docs was clearer.
And, I will create a separate PR for changes to `pluck`

Author: Supported by https://github.com/rails/rails/pull/48404/commits/900bf10890494ed1e3c6cc4f2ae05a5cbcdf6c6a

Changed the description to use some of the wording from the API documentation. Also, Added nuances to the main useful cases.

`pick` is a short-hand for `relation.limit(1).pluck(*column_names).first`, which makes it possible to replace code like:

> ```ruby
> Customer.limit(1).pluck(:id).first
> ```
> 
> with:
> 
> ```ruby
> Customer.pick(:id)
> ```

Reviewer: The example shows how easy it is to shorten the code with pick.
But I think it misses the nuance that pick "is primarily useful when you have a relation that’s already narrowed down to a single row.".

````suggestion
```ruby
Customer.where(id: 1).pluck(:id).first
```

with:

```ruby
Customer.where(id: 1).pick(:id)
```
````

Author: Indeed, an important nuance was missing. I will reflect the suggested source code.

Author: Supported by https://github.com/rails/rails/pull/48404/commits/9904db03670c49d3f75f21d6cb72f832974475a9

[`pluck`]: https://api.rubyonrails.org/classes/ActiveRecord/Calculations.html#method-i-pluck

### `pick`

> [`pick`][] can be used to pick the value(s) from the named column(s) in the current relation. It accepts a list of column names as an argument and returns the first one of the specified column values ​​with corresponding data type.

Reviewer: Last nitpick 😄 
```suggestion
[`pick`][] can be used to pick the value(s) from the named column(s) in the current relation. It accepts a list of column names as an argument and returns the first row of the specified column values ​​with corresponding data type.
```

Author: sounds good. Fixed here.

https://github.com/rails/rails/pull/48404/commits/b717293eb25a970a1b8342204a1a970d3dd0fd24
